src/CMakeLists.txt

```diff
@@ -15,13 +15,20 @@ SET(LIBGIT2_PC_LIBS "")
 ### Setup of library targets
 ###
 
-ADD_LIBRARY(git2internal OBJECT)
+# This is a workaround for old CMake versions which didn't yet
+# allow libraries without any source files.
```

Comment:
I'm somewhat surprised you need more dummies, for platforms others than macOS — that workaround is part of #4439, and it's unknown if recent versions of the Xcode generator work without it.